programs/client/Client.cpp

```diff
@@ -127,6 +127,69 @@ void Client::showWarnings()
     }
 }
 
+void Client::parseConnectionsCredentials()
+{
+    /// It is not possible to correctly handle multiple --host --port options.
+    if (hosts_and_ports.size() >= 2)
+        return;
+
+    String host;
+    std::optional<UInt16> port;
+    if (hosts_and_ports.empty())
+    {
+        host = config().getString("host", "localhost");
+        if (config().has("port"))
+            port = config().getInt("port");
+    }
+    else
+    {
+        host = hosts_and_ports.front().host;
+        port = hosts_and_ports.front().port;
+    }
+
+    Strings keys;
+    config().keys("connections_credentials", keys);
+    for (const auto & connection : keys)
+    {
+        const String & prefix = "connections_credentials." + connection;
+
+        const String & connection_name = config().getString(prefix + ".name", "");
+        if (connection_name != host)
+            continue;
+
+        String connection_hostname;
+        if (config().has(prefix + ".hostname"))
+            connection_hostname = config().getString(prefix + ".hostname");
+        else
+            connection_hostname = connection_name;
```

Comment:
It feels like in this case we should get connection host name from the root, e.g. from your test it would be:
```
<clickhouse>
     <host>$TEST_HOST</host>
```
Probably you did in the way it is now because connection_name is defined as `--host` and that is why it is probably better to do as in my first comment?

Reply:
The `connection_name` is equal to `--host` or `host` from the config here, so this is not important.

The idea was to override some settings for connecting to the specific host, basically this is for credentials, so that the user will not specify `--password` but simply `--host`. So the only setting that is required is the `name` or `hostname`.

About `--connection`, the reason for not introducing new settings are:
- do not let users teach new settings
- by analogy with `netrc`, but allowing override `hostname` (like `ssh_config` allows), this is mostly to give prettier name to IP addresses/auto-generated hotsnames

I don't have objections to introduce `--connection` setting, let me know if you did not change your mind.

Reply:
@kssenii After using this for a while, overriding host names too, indeed can be useful, here is a follow-up - #46480